source/common/http/http1/codec_impl.cc

```diff
@@ -259,7 +261,13 @@ const ToLowerTable& ConnectionImpl::toLowerTable() {
 }
 
 ConnectionImpl::ConnectionImpl(Network::Connection& connection, http_parser_type type)
-    : connection_(connection) {
+    : connection_(connection), output_buffer_(Buffer::InstancePtr{new Buffer::OwnedImpl()},
+                                              [&]() -> void { this->onBelowLowWatermark(); },
+                                              [&]() -> void { this->onAboveHighWatermark(); }) {
+  uint32_t buffer_limit = connection.bufferLimit();
+  if (buffer_limit > 0) {
+    output_buffer_.setWatermarks(buffer_limit / 2, buffer_limit);
```

Comment:
Fair enough. I don't have a strong opinion. I was mostly wondering if the code for this should be in one place at least for codecs in a utility helper? Then there could be a comment? I guess I would have a small preference for that.